File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = 0, sell = 1, max_profit = 0;

        while (sell < prices.size()) {
            if (prices[sell] < prices[buy]) {
                buy = sell;
            } else {
                max_profit = std::max(max_profit, prices[sell] - prices[buy]);
            }
            sell++;
        }
        
        return max_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy, sell, profit = 0, 1, 0
        
        while sell < len(prices):
            if prices[sell] < prices[buy]:
                buy = sell
            else:
                profit = max(profit, prices[sell] - prices[buy])
            sell += 1
        
        return profit